### ISSUE TITLE
Allow running 'projen' in a subdirectory

### ISSUE THREAD

**Issue by rix0rrr** [projen/projen#2418]
I would be helpful if `npx projen` would be able to run up the project tree and find `.projenrc.js`/`.projenrc.ts` in a parent directory and run itself there.

Current behavior:

```
$ npx projen
Error: Cannot find module '/path/to/project/some/subdirectory/.projenrc.js'
```

**Comment by fynnfluegge** [comment]
I would appreciate this feature, hence I tried to implement this in `src/cli/index.ts`:

```typescript
const DEFAULT_RC = findProjenRc(process.cwd());

...


/**
 * Run up project tree to find .projenrc.js
 *
 * @param cwd current working directory
 * @returns path to .projenrc.js or undefined if not found
 */
function findProjenRc(cwd: string): string | undefined {
  if (cwd === "/") {
    return undefined;
  }

  const projenrc = resolve(cwd, PROJEN_RC);
  if (fs.existsSync(projenrc)) {
    return projenrc;
  }

  return findProjenRc(resolve(cwd, ".."));
}
```

After `yarn link` my local dev projen and hitting `pj default` in a subdirectory of a projen project I'm still getting
```console
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '/path/to/project/some/subproject/.projenrc.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

Any idea what I am missing? 💡

**Comment by mrgrain** [comment]
Do you have branch/PR to look at?

Couple things:

- You probably don't want to mess around with `yarn link` etc. Instead add a test case that sets up the directory structure. We already have helpers to run projen cli. See: https://github.com/projen/projen/blob/53839b8d9459c9e7c2d4d2cd176ca7480d809138/test/cli.test.ts
- Not sure of overloading the `--rc` option makes sense here. It's marked as a legacy option ([reference](https://github.com/projen/projen/blob/e5899dd04a575209424a08fe90bde99e07ac6c7b/src/cli/synth.ts#L76)). Maybe instead we should try and find the closest `PROJEN_DIR`

**Comment by fynnfluegge** [comment]
Thanks for the hint! I added a tests and it's passing ✅ This is my [branch](https://github.com/fynnfluegge/projen/tree/feat/cli_find_projenrc).

**Comment by fynnfluegge** [comment]
Yes I had the feeling that my changes were not sufficient since with `yarn link` it didn't work. I will adjust my changes with your suggestion to look for `.projen/tasks.json`and raise PR then ✅

**Comment by fynnfluegge** [comment]
@mrgrain Raised a PR https://github.com/projen/projen/pull/2493. Got it running to run tasks in subdirectories with `yarn link` ✅
